test/core/test/mocked.test.ts

```diff
@@ -134,6 +135,130 @@ test('async functions should be mocked', () => {
   expect(asyncFunc()).resolves.toBe('foo')
 })
 
+describe('mocked function which fails on toReturnWith', () => {
+  test('zero call', () => {
+    const mock = vi.fn(() => 1)
+    try {
```

Comment:
Lets' rewrite these tests with `toThrow`:

```ts
expect(() => expect(mock).toReturnWith(2)).toThrowErrorMatchingInlineSnapshot()
```

Two reasons:

- 1) less code
- 2) more reliable code (if first assertion won't fail, we will not make assertions inside a "catch", but the test will pass)